server/interpreter/testcases.js

```diff
@@ -0,0 +1,460 @@
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Test cases for JavaScript interpreter.
+ * @author cpcallen@google.com (Christopher Allen)
+ */
+
+// This file is input for (and required by) gentests.js.
+exports.tests = [
+  { name: "onePlusOne", src: `
+    1 + 1;
+    `,
+    expected: 2 },
+
+  { name: "twoPlusTwo", src: `
+    2 + 2;
+    `,
+    expected: 4 },
+
+  { name: "sixTimesSeven", src: `
+    6 * 7;
+    `,
+    expected: 42 },
+
+  { name: "simpleFourFunction", src: `
+    (3 + 12 / 4) * (10 - 3);
+    `,
+    expected: 42 },
+
+  { name: "variableDecl", src: `
+    var x = 43;
+    x;
+    `,
+    expected: 43 },
+
+  { name: "condTrue", src: `
+    true?"then":"else";
+    `,
+    expected: "then" },
+
+  { name: "condFalse", src: `
+    false?"then":"else";
+    `,
+    expected: "else" },
+
+  { name: "ifTrue", src: `
+    if(true) {
+        "then";
+    }
+    else {
+        "else";
+    }
+    `,
+    expected: "then" },
+
+  { name: "ifFalse", src: `
+    if(false) {
+        "then";
+    }
+    else {
+        "else";
+    }
+    `,
+    expected: "else" },
+
+  { name: "simpleAssignment", src: `
+    var x = 0;
+    x = 44;
+    x;
+    `,
+    expected: 44 },
+
+  { name: "propertyAssignment", src: `
+    var o = {};
+    o.foo = 45;
+    o.foo;
+    `,
+    expected: 45 },
+
+  { name: "postincrement", src: `
+    var x = 45;
+    x++;
+    x++;
+    `,
+    expected: 46 },
+
+  { name: "preincrement", src: `
+    var x = 45;
+    ++x;
+    ++x;
+    `,
+    expected: 47 },
+
+  { name: "concat", src: `
+    "foo"+"bar";
+    `,
+    expected: "foobar" },
+
+  { name: "plusequalsLeft", src: `
+    var x=40, y=8;
+    x+=y;
+    x;
+    `,
+  expected: 48 },
+
+  { name: "plusequalsRight", src: `
+    var x=40, y=8;
+    x+=y;
+    y;
+    `,
+    expected: 8 },
+
+  { name: "simpleFunctionExpression", src: `
+    var v;
+    var f = function() {v = 49};
+    f();
+    v;
+    `,
+    expected: 49 },
+
+  { name: "fExpWithParameter", src: `
+    var v;
+    var f = function(x) {v = x};
+    f(50);
+    v;
+    `,
+    expected: 50 },
+
+  { name: "functionWithReturn", src: `
+    (function(x) {return x})(51);
+    `,
+    expected: 51 },
+
+  { name: "functionWithoutReturn", src: `
+    (function() {})();
+    `,
+    expected: undefined },
+
+  { name: "multipleReturn", src: `
+    var f = function() {
+        try {
+            return true;
+        }
+        finally {
+            return false;
+        }
+    }
+    f();
+    `,
+    expected: false },
+
+  { name: "throwCatch", src: `
+    var f = function () {
+        throw 26;
+    }
+    try {
+        f();
+    }
+    catch (e) {
+        e * 2;
+    }
+    `,
+    expected: 52 },
+
+  { name: "seqExpr", src: `
+    51, 52, 53;
+    `,
+    expected: 53 },
+
+  { name: "labeledStatement", src: `
+    foo: 54;
+    `,
+    expected: 54 },
+
+  { name: "whileLoop", src: `
+    var a = 0;
+    while(a<55) {
```

Comment:
Spaces needed before open parens.  Seven instances.
Spaces needed on either side of < operator.